pipelines/build/common/openjdk_build_pipeline.groovy

```diff
@@ -906,28 +905,16 @@ class Build {
     Executed on a build node, the function checks out the repository and executes the build via ./make-adopt-build-farm.sh
     Once the build completes, it will calculate its version output, commit the first metadata writeout, and archive the build results.
     */
-    def buildScripts(cleanWorkspace, filename) {
+    def buildScripts(cleanWorkspace, cleanWorkspaceAfter, cleanWorkspaceBuildOutputAfter, filename) {
         return context.stage("build") {
-
             if (cleanWorkspace) {
                 try {
 
                     try {
                         context.timeout(time: buildTimeouts.NODE_CLEAN_TIMEOUT, unit: "HOURS") {
-                            if (buildConfig.TARGET_OS == "windows") {
-                                // Windows machines struggle to clean themselves, see:
-                                // https://github.com/AdoptOpenJDK/openjdk-build/issues/1855
-                                context.sh(script: "rm -rf C:/workspace/openjdk-build/workspace/build/src/build/*/jdk/gensrc")
-                                // https://github.com/AdoptOpenJDK/openjdk-infrastructure/issues/1419
-                                context.sh(script: "rm -rf J:/jenkins/tmp/workspace/build/src/build/*/jdk/gensrc")
-                                // https://github.com/AdoptOpenJDK/openjdk-infrastructure/issues/1662
-                                context.sh(script: "rm -rf E:/jenkins/tmp/workspace/build/src/build/*/jdk/gensrc")
-                                // https://github.com/AdoptOpenJDK/openjdk-infrastructure/issues/1818
-                                context.sh(script: "rm -rf C:/Jenkins/temp/workspace/build/src/build/*/jdk/gensrc")
-                                context.cleanWs notFailBuild: true, disableDeferredWipeout: true, deleteDirs: true
-                            } else {
-                                context.cleanWs notFailBuild: true
-                            }
+                            // Note: Underlying org.apache DirectoryScanner used by cleanWs has a bug scanning where it misses files containing ".." so use rm -rf instead
+                            context.println "Cleaning workspace files: " + context.WORKSPACE + "/*"
+                            context.sh(script: "rm -rf " + context.WORKSPACE + "/*") 
```

Comment:
Is the same result achieved if you remove the `/*`? i.e just rm the workspace? I am just worried if WORKSPACE ever comes back empty.

Reply:
I was thinking about that, and I see your point, I hadn't thought about it that way round. Without the /* it will remove the workspace folder as well, which I think will cause a problem. I can add a check that WORKSPACE is not empty...

Reply:
Ok that's probably a good idea. You might be right removing the folder might not work. We have a cleanup job that runs to rm all our workspaces but it ignores the cleanup job folder even though there's nothing in it. I think the pipeline complains.